File: 백준/Silver/1181.단어 정렬/단어 정렬.py
N = int(input())
lst = []
for _ in range(N):
    lst.append(input())

lst.sort()
lst.sort(key=lambda x:len(x))

i = 0
while i < len(lst) -1:
    if lst[i] == lst[i+1]:
        lst.pop(i+1)
        continue
    i += 1

for i in lst:
    print(i)

File: 백준/Silver/1181. 단어 정렬/단어 정렬.js
function solution(input) {
  const [_, ...list] = input;

  const itemSet = new Set(list);
  const lenSet = new Set([]);
  const map = {};

  [...itemSet].forEach((item) => {
    const l = String(item.length);

    if (lenSet.has(l)) {
      map[l].push(item);
    } else {
      map[l] = [item];
    }

    lenSet.add(l);
  });

  let result = [];
  const lenArr = [...lenSet].map((v) => +v).sort((a, b) => a - b);
  lenArr.forEach((l) => {
    const subArr = map[String(l)].sort((a, b) => (a > b ? 1 : -1));
    result = result.concat(subArr);
  });

  return result;
}

(function main() {
  let isLocal = false;
  let fs = require("fs");
  let filePath = isLocal ? "t.txt" : "/dev/stdin";
  let input = fs.readFileSync(filePath).toString().trim().split("\n");

  const answer = solution(input).join("\n");
  console.log(answer);
})();
